manylinux_v2_centos/Dockerfile

```diff
@@ -0,0 +1,95 @@
+
+ENV PYENV_ROOT="/pyenv" \
+    PATH="/pyenv/bin:/pyenv/shims:$PATH"
+
+RUN yum update -y && yum install -y epel-release wget gcc zlib-devel bzip2 bzip2-devel readline-devel sqlite sqlite-devel xz xz-devel libffi-devel curl git ncurses-devel numactl numactl-devel openssh-clients libcudnn8-devel zip blas-devel lapack-devel protobuf-compiler autoconf automake libtool centos-release-scl scl-utils cmake && yum clean all
+
+# Install tools that have to be installed after the previous install completes.
+RUN yum remove -y git && yum install -y https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm && yum install -y git jq devtoolset-11 && yum remove -y endpoint-repo
+
+# Install latest openssl (can't use the package manager because CentOS 7 stopped after openssl 1.0)
+RUN pushd tmp && \
+    wget https://ftp.openssl.org/source/openssl-1.1.1k.tar.gz && \
+    tar -xzvf openssl-1.1.1k.tar.gz &&\
+    cd openssl-1.1.1k && \
+    ./config --prefix=/usr --openssldir=/etc/ssl --libdir=lib no-shared zlib-dynamic && \
+    make && \
+    make install && \
+    popd
+
+# Install sccache
+ARG SCCACHE_VERSION=0.4.2
+ARG ARCH="x86_64"
+
+RUN curl -o /tmp/sccache.tar.gz \
+        -L "https://github.com/mozilla/sccache/releases/download/v${SCCACHE_VERSION}/sccache-v${SCCACHE_VERSION}-${ARCH}-unknown-linux-musl.tar.gz" && \
+        tar -C /tmp -xvf /tmp/sccache.tar.gz && \
+        mv "/tmp/sccache-v${SCCACHE_VERSION}-${ARCH}-unknown-linux-musl/sccache" /usr/bin/sccache && \
+        chmod +x /usr/bin/sccache
+
+# Install ucx
+ARG UCX_VERSION=1.13.0
+RUN mkdir -p /ucx-src && pushd /ucx-src &&\
+    git clone https://github.com/openucx/ucx -b v${UCX_VERSION} ucx-git-repo &&\
+    cd ucx-git-repo && \
+    ./autogen.sh && \
+    ./contrib/configure-release \
+       --prefix=/usr               \
+       --enable-mt                 \
+       --enable-cma                \
+       --enable-numa               \
+       --with-gnu-ld               \
+       --with-sysroot              \
+       --without-verbs             \
+       --without-rdmacm            \
+       --with-cuda=/usr/local/cuda && \
+    CPPFLAGS=-I/usr/local/cuda/include make -j && \
+    make install && \
+    popd && \
+    rm -rf /ucx-src/
+
+# for pyenv
+ARG PY39_VERSION="3.9.16"
+ARG PY310_VERSION="3.10.9"
+
+# Install pyenv
+RUN curl https://pyenv.run | bash
+
+# Create pyenvs
+# TODO: Determine if any cleanup of the pyenv layers is needed to shrink the container
+RUN pyenv update
+
+# Need to specify the openssl location because of the install from source
+RUN CPPFLAGS="-I/usr/include/openssl" LDFLAGS="-L/usr/lib" pyenv install --verbose ${PY39_VERSION}
+
+RUN CPPFLAGS="-I/usr/include/openssl" LDFLAGS="-L/usr/lib" pyenv install --verbose ${PY310_VERSION}
+
+RUN pyenv global ${PY39_VERSION} && python -m pip install auditwheel patchelf twine && pyenv rehash
+
+RUN pyenv global ${PY310_VERSION} && python -m pip install auditwheel patchelf twine && pyenv rehash
```

Comment:
I'm not super familiar with these images, but it appears we're adding Python `3.9` and `3.10` environments to a single image variant here.

Is that accurate? If so, does it make sense to have one image per Python version?

Then we could control the Python versions at the GHAs matrix level, which is consistent with our other images.